src/main/java/org/commcare/util/EncryptionUtils.java

```diff
@@ -1,49 +1,117 @@
 package org.commcare.util;
 
+import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.charset.Charset;
 import java.security.InvalidAlgorithmParameterException;
 import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.KeyStoreException;
 import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
 
 import javax.crypto.BadPaddingException;
 import javax.crypto.Cipher;
 import javax.crypto.IllegalBlockSizeException;
 import javax.crypto.NoSuchPaddingException;
-import javax.crypto.SecretKey;
 import javax.crypto.spec.GCMParameterSpec;
-import javax.crypto.spec.IvParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
 
 public class EncryptionUtils {
 
+    public static final String USER_CREDENTIALS_KEY_ALIAS = "user-credentials-key-alias";
+
+    public enum CryptographicOperation {Encryption, Decryption}
+
+    private static IEncryptionKeyProvider encryptionKeyProvider = EncryptionKeyServiceProvider.getInstance().serviceImpl();
+
+    public static IEncryptionKeyProvider getEncryptionKeyProvider() {
```

Comment:
I like the idea of having a helper class that can be used to instantiate the appropriate `EncryptionKeyProvider`, this would mean dropping the SPI mechanism and sourcing the Key Store name from some other way (I suggested using CommCarePlatform before, maybe we can revisit that option, see [this](https://github.com/dimagi/commcare-core/pull/1349#discussion_r1364609057) comment).

Reply:
Yes, that makes sense to me.